src/app/public/modules/file-attachment/file-attachment.component.ts

```diff
@@ -0,0 +1,352 @@
+    this.labelComponents.changes
+    .takeUntil(this.ngUnsubscribe)
+    .subscribe(() => {
+      this.changeDetector.detectChanges();
+    });
+  }
+
+  public isImage(): boolean {
+    return this.fileItemService.isImage(this.value);
+  }
+
+  public onDropClicked(): void {
+    this._onTouched();
+    this.inputEl.nativeElement.click();
+  }
+
+  public fileChangeEvent(fileChangeEvent: any): void {
+    this.handleFiles(fileChangeEvent.target.files);
+  }
+
+  public fileDragEnter(dragEnterEvent: any): void {
+    // Save this target to know when the drag event leaves
+    this.enterEventTarget = dragEnterEvent.target;
+    dragEnterEvent.stopPropagation();
+    dragEnterEvent.preventDefault();
+  }
+
+  public fileDragOver(dragOverEvent: any): void {
+    const transfer = dragOverEvent.dataTransfer;
+
+    dragOverEvent.stopPropagation();
+    dragOverEvent.preventDefault();
+
+    if (transfer) {
+      if (transfer.items) {
+        const files = transfer.items;
+
+        for (let index = 0; index < files.length; index++) {
+          const file: any = files[index];
+
+          if (file.type && this.fileAttachmentService.fileTypeRejected(file.type, this.acceptedTypes)) {
+            this.rejectedOver = true;
+            this.acceptedOver = false;
+            return;
+          }
+        }
+
+        if (files.length > 0 && !this.acceptedOver) {
+          this.rejectedOver = false;
+          this.acceptedOver = true;
+        }
+
+      } else if (transfer.files) {
+        // If the browser does not support DataTransfer.items,
+        // defer file-type checking to drop handler.
+        // https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/items#Browser_compatibility
+        this.rejectedOver = false;
+        this.acceptedOver = true;
+      }
+    }
+  }
+
+  public fileDrop(dropEvent: any): void {
+    dropEvent.stopPropagation();
+    dropEvent.preventDefault();
+
+    this.enterEventTarget = undefined;
+    this.rejectedOver = false;
+    this.acceptedOver = false;
+
+    if (dropEvent.dataTransfer && dropEvent.dataTransfer.files) {
+      if (this.fileAttachmentService.verifyDropFiles(dropEvent.dataTransfer.files)) {
+        this.handleFiles(dropEvent.dataTransfer.files);
+      }
+    }
+  }
+
+  public fileDragLeave(dragLeaveEvent: any): void {
+    if (this.enterEventTarget === dragLeaveEvent.target) {
+      this.rejectedOver = false;
+      this.acceptedOver = false;
+    }
+  }
+
+  public deleteFileAttachment(): void {
+    this.value = undefined;
+    this.changeDetector.markForCheck();
+    this.emitFileChangeEvent(this.value);
+  }
+
+  public getFileName(): string | undefined {
+    if (this.value) {
+      // tslint:disable-next-line: max-line-length
+      let dropName = this.fileItemService.isFile(this.value) && this.value.file.name ? this.value.file.name : this.value.url;
+
+      if (dropName.length > 26) {
+        return dropName.slice(0, 26) + '....';
```

Comment:
Typically an ellipsis is only 4 dots when at the end of a sentence. I'd change this to 3 dots.

Reply:
this was in the design, so should I clarify with Todd?

Reply:
Should use 3 dots as in text expand.

Reply:
Yeah, it’s probably worth getting Todd to take a look.

Reply:
I feel lied to.